src/SourceDocumentCoordinator/SourceDocumentCoordinator/Handlers/GetForwardDocumentLinksCommandHandler.cs

```diff
@@ -13,42 +19,116 @@ public class GetForwardDocumentLinksCommandHandler : IHandleMessages<GetForwardD
     {
         private readonly WorkflowDbContext _dbContext;
         private readonly IDataServiceApiClient _dataServiceApiClient;
+        private readonly ILogger<GetForwardDocumentLinksCommandHandler> _logger;
 
-        public GetForwardDocumentLinksCommandHandler(WorkflowDbContext dbContext, IDataServiceApiClient dataServiceApiClient)
+        public GetForwardDocumentLinksCommandHandler(
+            WorkflowDbContext dbContext,
+            IDataServiceApiClient dataServiceApiClient,
+            ILogger<GetForwardDocumentLinksCommandHandler> logger)
         {
             _dbContext = dbContext;
             _dataServiceApiClient = dataServiceApiClient;
+            _logger = logger;
         }
+
         public async Task Handle(GetForwardDocumentLinksCommand message, IMessageHandlerContext context)
         {
+            LogContext.PushProperty("MessageId", context.MessageId);
+            LogContext.PushProperty("Message", message.ToJSONSerializedString());
+            LogContext.PushProperty("EventName", nameof(GetForwardDocumentLinksCommand));
+            LogContext.PushProperty("MessageCorrelationId", message.CorrelationId);
+            LogContext.PushProperty("ProcessId", message.ProcessId);
+            LogContext.PushProperty("SourceDocumentId", message.SourceDocumentId);
+
+            _logger.LogInformation("Entering {EventName} handler with: {Message}");
+
             var docLinks = await _dataServiceApiClient.GetForwardDocumentLinks(message.SourceDocumentId);
+
+            if (docLinks == null || docLinks.Count == 0)
+            {
+                _logger.LogInformation("No Forward linked documents found for SourceDocumentId: {SourceDocumentId}");
+
+                return;
+            }
+
             var linkedDocIds = docLinks.Select(s => s.DocId2).ToList();
 
-            if (linkedDocIds?.Count == 0) return;
+            LogContext.PushProperty("LinkedDocuments", linkedDocIds.ToJSONSerializedString());
+
+            _logger.LogInformation("Forward linked documents found for SourceDocumentId: {SourceDocumentId}; LinkedDocuments: {LinkedDocuments}");
+
+            var failedLinkedDocuments = new StringBuilder(linkedDocIds.Count);
+
+            await PersistLinkedDocuments(message, linkedDocIds, failedLinkedDocuments);
+
+            if (failedLinkedDocuments.Length > 0)
+            {
+                throw new ApplicationException(
+                    $"Errors while adding Backward linked documents to SourceDocumentId: {message.SourceDocumentId}:" +
+                    $"{Environment.NewLine}{failedLinkedDocuments}");
+            }
+
+            _logger.LogInformation("Successfully Completed Event {EventName}: {Message}");
+        }
 
+        private async Task PersistLinkedDocuments(GetForwardDocumentLinksCommand message, IEnumerable<int> linkedDocIds,
```

Comment:
Could we start using Async in name as convention to make intention clearer?
```suggestion
        private async Task PersistLinkedDocumentsAsync(GetForwardDocumentLinksCommand message, IEnumerable<int> linkedDocIds,
```